src/Orc.Extensibility/Models/CosturaRuntimeAssembly.cs

```diff
@@ -0,0 +1,144 @@
+
+        public string ResourceName { get; set; }
+
+        public string Version { get; set; }
+
+        public string AssemblyName { get; set; }
+
+        public string RelativeFileName { get; set; }
+
+        public long? Size { get; set; }
+
+        public EmbeddedResource EmbeddedResource { get; set; }
+
+        public override bool IsRuntime
+        {
+            get
+            {
+                var resourceName = ResourceName;
+                if (!string.IsNullOrWhiteSpace(resourceName))
+                {
+                    return resourceName.ContainsIgnoreCase(".runtimes.");
+                }
+
+                return EmbeddedResource?.Name.ContainsIgnoreCase(".runtimes.") ?? false;
+            }
+        }
+
+        public override Stream GetStream()
+        {
+            if (_cachedData is null)
+            {
+                // Note: we preferred not to cache, but we can't read the same stream twice it seems
+                var embeddedResource = EmbeddedResource;
+
+                unsafe
+                {
+                    using (var resourceStream = new UnmanagedMemoryStream(embeddedResource.Start, embeddedResource.Size))
+                    {
+                        using (var stream = LoadStream(resourceStream, embeddedResource.Name))
+                        {
+                            _cachedData = ReadStream(stream);
+                        }
+                    }
+                }
+            }
+
+            return new MemoryStream(_cachedData);
+        }
+
+        public override string ToString()
+        {
+            return $"{ResourceName}|{Version}|{AssemblyName}|{RelativeFileName}|{Checksum}|{Size}";
+        }
+
+#pragma warning disable IDISP015 // Member should not return created and cached instance
+        private Stream LoadStream(Stream existingStream, string resourceName)
+#pragma warning restore IDISP015 // Member should not return created and cached instance
+        {
+            if (resourceName.EndsWith(".compressed"))
+            {
+                var originalPosition = existingStream.Position;
+
+                using (var source = new DeflateStream(existingStream, CompressionMode.Decompress))
+                {
+                    var memoryStream = new MemoryStream();
+
+                    CopyTo(source, memoryStream);
+
+                    memoryStream.Position = 0L;
+                    existingStream.Position = originalPosition;
+
+                    return memoryStream;
+                }
+
+            }
+
+            return existingStream;
+        }
+
+        private void CopyTo(Stream source, Stream destination)
+        {
+            var array = new byte[81920];
+            int count;
+
+            while ((count = source.Read(array, 0, array.Length)) != 0)
+            {
+                destination.Write(array, 0, count);
+            }
+
+            destination.Flush();
+        }
+
+        private byte[] ReadStream(Stream stream)
```

Comment:
Is there any profit for hiding MemoryStream under Stream reference then excplicitly create an array instead of using ToArray()?

Reply:
I am not sure I understand this comment. The source stream is not a memory stream?

Reply:
If i understood correctly it's known stream passed to ReadStream is always MemoryStream. In this way you can use ToArray